tests/compile_pip_requirements/BUILD.bazel

```diff
@@ -32,3 +32,33 @@ compile_pip_requirements(
     requirements_in = "requirements.txt",
     requirements_txt = "requirements_lock.txt",
 )
+
+genrule(
+    name = "generate_os_specific_requirements_in",
+    srcs = [],
+    outs = ["requirements_os_specific.in"],
+    cmd = """
+cat > $@ <<EOF
+pip==22.3.0 ; sys_platform == "linux"
+pip==22.2.2 ; sys_platform == "darwin"
+pip==22.2.1 ; sys_platform == "win32"
```

Comment:
nit: currently `pip_parse` does not support platform specifiers as described in #1105 and the output os specific requirements files will not be parseable by `pip_parse`.

However, since this is a test and not an example, it is fine.